### PR TITLE
Map

### PR DESCRIPTION
**Pull request by AdamBeresnev**

### CODE REVIEW

File: WebService/ComparisonEngineUI/ComparisonEngineUI.Android/ComparisonEngineUI.Android.csproj [hunk 1]
```diff
@@ -32,6 +32,12 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <AndroidLinkMode>None</AndroidLinkMode>
+    <AotAssemblies>false</AotAssemblies>
+    <EnableLLVM>false</EnableLLVM>
+    <AndroidEnableProfiledAot>false</AndroidEnableProfiledAot>
+    <BundleAssemblies>false</BundleAssemblies>
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI.Android/ComparisonEngineUI.Android.csproj:38]: Does this part need to repeat in line 51-54?

File: WebService/ComparisonEngineUI/ComparisonEngineUI.Android/ComparisonEngineUI.Android.csproj [hunk 2]
```diff
@@ -99,7 +112,7 @@
   <Import Project="$(MSBuildExtensionsPath)\Xamarin\Android\Xamarin.Android.CSharp.targets" />
   <ProjectExtensions>
     <VisualStudio>
-      <UserProperties TriggeredFromHotReload="True" XamarinHotReloadDebuggerTimeoutExceptionComparisonEngineUIAndroidHideInfoBar="True" />
+      <UserProperties TriggeredFromHotReload="False" XamarinHotReloadDebuggerTimeoutExceptionComparisonEngineUIAndroidHideInfoBar="True" />
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI.Android/ComparisonEngineUI.Android.csproj:115]: why change this to false?

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Constants.cs [hunk 3]
```diff
@@ -6,6 +6,22 @@ namespace ComparisonEngineUI
...
+        public static string GoogleScope = "https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/userinfo.profile";
+        public static string GoogleAuthorizeUrl = "https://accounts.google.com/o/oauth2/auth";
+        public static string GoogleAccessTokenUrl = "https://www.googleapis.com/oauth2/v4/token";
+        public static string GoogleUserInfoUrl = "https://www.googleapis.com/oauth2/v2/userinfo";
+
+        // Set these to reversed iOS/Android client ids, with :/oauth2redirect appended
+        public static string GoogleiOSRedirectUrl = "<insert IOS redirect URL here>:/oauth2redirect";
+        public static string GoogleAndroidRedirectUrl = "com.googleusercontent.apps.433871991239-rq3kn77qftb5qndip967jdb5nbgjebra:/oauth2redirect";
+
+
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI/Constants.cs:24]: Damn, these should probably go to secrets or be omitted from GH, no?
also one whitespace at the end too much

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/MapPage.xaml.cs [hunk 4]
```diff
@@ -6,16 +6,90 @@
...
+using ComparisonEngineUI.Models;
 
 namespace ComparisonEngineUI.Views
 {
     [XamlCompilation(XamlCompilationOptions.Compile)]
     public partial class MapPage : ContentPage
     {
+        Map map;
+        Xamarin.Essentials.Location currentPosition;
         public MapPage()
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI/Views/MapPage.xaml.cs:20]: shouldn't there be a whitespace between these?

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/MapPage.xaml.cs [hunk 5]
```diff
@@ -6,16 +6,90 @@
...
+
+        public void SetBarDirections(Bar bar)
+        {
+            SetBarPin(bar);
+        }
+
+        public async void SetBarPin(Bar bar)
+        {
+            var location = (await Xamarin.Essentials.Geocoding.GetLocationsAsync(bar.BarLocation)).FirstOrDefault();
+            if (location != null)
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI/Views/MapPage.xaml.cs:67]: probably need a whitespace between these

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/MapPage.xaml.cs [hunk 6]
```diff
@@ -6,16 +6,90 @@
...
+            Content = stack;
+        }
+
+        public void SetDrinkPins(List<Bar> items)
+        {
+            foreach (Bar item in items)
+            {
+                SetBarPin(item);
+            }
+        }
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI/Views/MapPage.xaml.cs:57]: not sure if this should be a List<Bar>, but can't think of anything better at this point

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/MapPage.xaml.cs [hunk 7]
```diff
@@ -6,16 +6,90 @@
...
+
+        public async void GetCurrentPosition()
+        {
+            try
+            {
+                currentPosition = await Xamarin.Essentials.Geolocation.GetLastKnownLocationAsync();
+            }
+            catch (Exception ex)
+            {
+                // Unable to get location
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI/Views/MapPage.xaml.cs:91]: shouldn't this at least throw? seems sus to have it just empty

**Review by ElektrosStulpas — COMMENTED** [review]
Not much of a code review, but it is what it is

File: WebService/ComparisonEngineUI/ComparisonEngineUI.Android/ComparisonEngineUI.Android.csproj (hunk 2 above)
Comment (AdamBeresnev) [reply WebService/ComparisonEngineUI/ComparisonEngineUI.Android/ComparisonEngineUI.Android.csproj:115]: I didn't change it. I think VSdid. Not sure what's the difference

File: WebService/ComparisonEngineUI/ComparisonEngineUI.Android/ComparisonEngineUI.Android.csproj (hunk 1 above)
Comment (AdamBeresnev) [reply WebService/ComparisonEngineUI/ComparisonEngineUI.Android/ComparisonEngineUI.Android.csproj:38]: Weird, it got autogenerated this way

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/MapPage.xaml.cs (hunk 7 above)
Comment (AdamBeresnev) [reply WebService/ComparisonEngineUI/ComparisonEngineUI/Views/MapPage.xaml.cs:94]: Don't think that's necessary. If an exception occurs at this point it does not negatively impact the app. If I'm not mistaken an exception is thrown here if the device does not have a last known location, in which case the map centers on the center of Vilnius

**Pull request merged**